ggml/examples/unity/model_loader.cpp

```diff
@@ -39,12 +39,15 @@ std::int64_t
 model_loader::load_model_weights(fairseq2_model &model, std::ifstream &fin)
 {
     std::int64_t num_tensor = 0;
-    std::int64_t ctx_size = 0;
+    std::int64_t f32_tensor_size = 0;
     fin.read((char*) &num_tensor, sizeof(num_tensor));
-    fin.read((char*) &ctx_size, sizeof(ctx_size));
+    fin.read((char*) &f32_tensor_size, sizeof(f32_tensor_size));
 
+    // TODO: it might be interesting to allow the caller to not upcast the weights to float32.
+    // Note this require changing the on disk format
+    bool as_float32 = true;
```

Comment:
Should we also allow the user to specify this as the func param ?

Reply:
there are more changes needed that just adding a flag, that's why I left it as a TODO.